### DISCUSSION TITLE
Setup failing between steps 3 and 4 on `/db`

### DISCUSSION THREAD

**Discussion by juekr** [JulianPrieber/littlelink-custom#383] [Issues]
Platform: `Debian GNU/Linux 11`
Server type: `NGINX`
PHP version: `8.2`

[//]: # (Please make sure you have read the system requirements at https://llc.ovh/req before submitting your issue.)

Like others in this forum, my installation/configuration stops between steps 3 and 4 with a 403 on a URL ending with `/db`. Unlike others I have a nginx setup and after reading about the others' experiences, I could imagine that there is something failing with the htaccess or apache vhost configurations. But I have no idea how to tackle or even debug them with a nginx webserver (came preinstalled with my hoster). Any ideas on what to do to solve or further track down the problem?

**Comment by JulianPrieber** [comment]
I thought we fixed this issue a while ago.

You can skip the installation with `example.com/skip`.

**Comment by juekr** [comment]
`/skip` gives me a 404 as well – did I install an old version by accident?

**Comment by JulianPrieber** [comment]
Do other routes still work?

**Comment by juekr** [comment]
Like?

**Comment by JulianPrieber** [comment]
Like just example.com or example.com/mysql-test.

**Comment by JulianPrieber** [comment]
During the installation, the app is set up to redirect all unknown routes to the home/installer page.
This means during setup, 404s shouldn't even be possible.

It will probably be best to reinstall the app.

The thing that confuses me is all that the `/db` route does is redirect you to the next step in the installer:

```
        if($request->database == 'MySQL'){
            return redirect(url('?mysql'));
        }else{
            return redirect(url('?4'));
        }
```

**Comment by juekr** [comment]
`/` works and starts the config process again, `/mysql-test` produces another 404. 

Did I maybe miss a step in the installation? Isn't it just "Download and extract the zip file"? Is there more to it?

**Comment by JulianPrieber** [comment]
On NGINX, you'll have to configure your block first for the app to function properly, but essentially yes.
https://laravel.com/docs/9.x/deployment#nginx
https://docs.littlelink-custom.com/getting-started/other-webservers/

One thing that's also important is that PHP is able to read and write to all the files, but in most cases this is possible by default.

**Comment by JulianPrieber** [comment]
It's nothing really different from installing WordPress.

**Comment by jmdv-es** [comment]
Hi @JulianPrieber, maybe I can share some light:

this error described by @juekr is also happening to me. It's happening while deploying the files in an IONOS web server.

But, using an OVH web server, the applications install without problem.

**Reply by aubreyz** [reply]
Yup - exactly that

**Comment by crackscout123** [comment]
I got the same issue, but fixed it by enabling .thacces but then encounterd another error the installer is looking for two file which dont exsist called "INSTALLING" and "INSTALLERLOCK" here is the error code:
 https://flareapp.io/share/W7zRk3k7#F55

**Comment by crackscout123** [comment]
Im going to reinstall it on a different server cause I changed my provider. I'll poste an update when im done.

**Comment by DrMxrcy** [comment]
I am getting the same error on my docker install. I can install with a local IP, but the setup redirects the /db to HTTP and 404s

**Comment by JulianPrieber** [comment]
@DrMxrcy 

> I am getting the same error on my docker install. I can install with a local IP, but the setup redirects the /db to HTTP and 404s

If you're accessing the docker directly over the IP and port, then it works?

**Comment by crackscout123** [comment]
> Im going to reinstall it on a different server cause I changed my provider. I'll poste an update when im done.

It wont work, i gave up

**Comment by JulianPrieber** [comment]
**For everyone still wondering what's causing this:**

This is most likely a Vhost issue in your setup or reverse proxy. You need to set up your app to redirect everything to HTTPS.
Cloudflare has an option to do this for you, with no additional setup. We have documentation on this for NGINX reverse proxy or web. Everything else should be documented [here](https://laravel.com/docs/9.x/readme) or online somewhere.

If anyone is having this issue with docker, set up your instance using the IP and port directly without a reverse proxy. Then on the config page, enable force HTTPS.

For anyone else, just skip the setup and you should be fine. On the setup page, visit `/skip` this generates a default admin profile, and you can use the app like normal. Then on the config page, enable force HTTPS.

You can log in with:
`admin@admin.com`
`12345678`

**Reply by aubreyz** [reply]
Not a solution (given you closed the issue after this posting). It was already pointed out to you above that /skip does not work, (and even if it did work would have been a workaround not a bug fix for this specific problem - failure of the install at step 3 with a failed misdirect which you claim "shouldn't even be possible".

**Reply by JulianPrieber** [reply]
Still not applicable to your issue. Your issue is PHP related this was a vhost issue.

You may be experiencing two separate issues.

**Comment by naufdotal** [comment]
Hi,
can someone post the full nginx config with laravel rewrites to use this linkstack?
because I'm still getting error 404 not found on /login or /register page, my nginx vhost config:

```
server {
    server_name myserver.com;
    index index.php index.html;
    root /srv/www/myserver/public;

    location / {
    try_files $uri $uri/ /index.php?$query_string;
    }

    client_max_body_size 0;
    chunked_transfer_encoding on;

    location ~ \.php$ {
       try_files $uri =404;
	fastcgi_index   index.php;
        fastcgi_pass unix:/run/php/php8.0-fpm.sock;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        include fastcgi_params;
    }

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/myserver.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/myserver.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot


	location ~* \.(jpg|jpeg|gif|css|png|js|ico|html)$ {
		access_log off;
		expires max;
		log_not_found off;
	}

# removes trailing slashes (prevents SEO duplicate content issues)
	if (!-d $request_filename)
	{
		rewrite ^/(.+)/$ /$1 permanent;
	}

	# unless the request is for a valid file (image, js, css, etc.), send to bootstrap
	if (!-e $request_filename)
	{
		rewrite ^/(.*)$ /index.php?/$1 last;
		break;
	}


}
server {
    if ($host = myserver.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

    server_name myserver.com;

    listen 80;
    return 404; # managed by Certbot

}
```

thanks!

**Comment by aubreyz** [comment]
I still have this problem - which is making installation impossible.

After pressing next on https://links.domainxxx.co.uk/?3
I get to
https://links.domainxxx.co.uk/db

with response without any redirection
Internal Server Error
The server encountered an internal error or misconfiguration and was unable to complete your request.
Please contact the server administrator at to inform them of the time this error occurred, and the actions you performed just before this error.
More information about this error may be available in the server error log.
Additionally, a 500 Internal Server Error error was encountered while trying to use an ErrorDocument to handle the request.

If I start again with
https://links.domainxxx.co.uk/skip
I get the same server error as above in immediate response so that's not a route out of it
Is there some other way to skip the setup

I have tried reinstalling from scratch several times (on different domains or subdomains) all with the same problem.
All on an Ionos server (I mention that because Ionos - a pretty big provider so may be coincidental - was mentioned above)

Going back to https://links.domainxxx.co.uk/ gets back to the start on the setup without problem

I just figured out that after /db
I can go manually to https://domainxxx.co.uk/?4
However if I fill that in
I get redirected to
https://links.oboehannah.co.uk/create-admin
with the same 404 again

Not sure where to go from here. Happy to work with you on debugging this on a real live instance

**Reply by JulianPrieber** [reply]
Issue seems unrelated.

**Reply by aubreyz** [reply]
No idea what this means. It is an identical longstanding issue of failure to redirect at one point of the setup. Unrelated to what?

**Reply by JulianPrieber** [reply]
The steps to fixing your issue do not relate to the steps described here.
Furthermore, your issue description is different from what was described here.

**Reply by aubreyz** [reply]
How is it different. It is absolutely identical. - the description of the install misdirecting at exactly step 3 is almost word for word as is everything else. Don't worry I'll use other software, just as the others who have raised this issue have done -- having raised issues only to have them closed down. What a pity because you have clearly put a lot of love and effort into this work. Cheers.
